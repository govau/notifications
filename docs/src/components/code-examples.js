import React from 'react'
import styled, { css } from 'styled-components'
import SyntaxHighligher from 'react-syntax-highlighter'
import { StaticQuery, graphql } from 'gatsby'
import ReactMarkdown from 'react-markdown'

const linkCSS = css`
  line-height: 1.25;
  padding: 1.25rem 1.5rem;
  border-bottom: ${props =>
    props.active ? '0.35rem solid #007099' : '0.35rem solid #e0e0e0'};
  text-decoration: none;
  display: block;
  color: #007099;

  &:hover {
    color: #000;
    cursor: pointer;
  }
`

const Tabs = styled.div`
  margin-top: 1rem;
  border-top: 1px solid #cfcfcf;
`

const TabsNav = styled.div`
  border-bottom: 0.35rem solid #e0e0e0;
  width: 100%;
`

const Ul = styled.ul`
  margin: 0 0 -0.35rem;
  padding: 0;
  display: inline-block;
  line-height: 1.5;
  list-style-type: none;
`

const NavItem = styled.li`
  margin: 0;
  padding: 0;
  display: inline-block;
`

const NavLink = styled.label`
  ${linkCSS};
`

const TabContent = styled.div`
  display: ${props => (props.active ? 'visible' : 'none')};
`

export default class CodeExamples extends React.Component {
  state = {
    activeTab: 0,
  }

  handleClick = index => {
    this.setState({ activeTab: index })
  }

  filterTransformSortCodeSnippets = codeSnippets =>
    codeSnippets
      .filter(n => n.node.relativePath.startsWith(this.props.path))
      .map(n => n.node)
      .sort((a, b) => a.name.localeCompare(b.name))

  syntaxHighlighter = snippet => {
    const customStyle = { margin: 0 }
    if (snippet.extension && snippet.extension.toLowerCase() === 'md') {
      return (
        <ReactMarkdown
          source={snippet.content}
          renderers={{
            pre: React.Fragment,
            code: ({ value, language }) => (
              <SyntaxHighligher
                customStyle={customStyle}
                language={language}
                children={value}
              />
            ),
          }}
        />
      )
    }

    return (
      <SyntaxHighligher
        customStyle={customStyle}
        language={snippet.extension}
        children={snippet.content}
      />
    )
  }

  renderTabs = data => {
    const codeSnippets = this.filterTransformSortCodeSnippets(
      data.allCodeSamples.edges
    )
    return (
      <Tabs>
        <TabsNav>
          <Ul>
            {codeSnippets.map((s, i) => (
              <NavItem key={i}>
                <NavLink
                  onClick={() => this.handleClick(i)}
                  active={this.state.activeTab === i}
                >
                  {s.name}
                </NavLink>
              </NavItem>
            ))}
          </Ul>
        </TabsNav>

        {codeSnippets.map((s, i) => (
          <TabContent key={i} active={this.state.activeTab === i}>
            {this.syntaxHighlighter(s)}
          </TabContent>
        ))}
      </Tabs>
    )
  }

  render = () => (
    <StaticQuery
      query={graphql`
        query {
          allCodeSamples {
            edges {
              node {
                content
                extension
                relativePath
                name
              }
            }
          }
        }
      `}
      render={data => this.renderTabs(data)}
    />
  )
}
