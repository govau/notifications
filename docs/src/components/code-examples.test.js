import React from 'react'
import renderer from 'react-test-renderer'
import { CodeExamplesComponent } from './code-examples'
import mockQuery from './code-examples.test.fixture'

describe('CodeExamplesComponent', () =>
  it('renders correctly', () => {
    const tree = renderer
      .create(<CodeExamplesComponent data={mockQuery} path="sending-emails" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  }))
