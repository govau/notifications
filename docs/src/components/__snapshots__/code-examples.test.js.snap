// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeExamplesComponent renders correctly 1`] = `
<div
  className="sc-bdVaJa hPuEoI"
>
  <div
    className="sc-bwzfXH hyVNwM"
  >
    <ul
      className="sc-htpNat bHRdbh"
    >
      <li
        className="sc-bxivhb fpojHh"
      >
        <label
          className="sc-ifAKCX ffLiWg"
          onClick={[Function]}
        >
          cURL
        </label>
      </li>
      <li
        className="sc-bxivhb fpojHh"
      >
        <label
          className="sc-ifAKCX iLlAOV"
          onClick={[Function]}
        >
          Java
        </label>
      </li>
      <li
        className="sc-bxivhb fpojHh"
      >
        <label
          className="sc-ifAKCX iLlAOV"
          onClick={[Function]}
        >
          Python
        </label>
      </li>
    </ul>
  </div>
  <div
    className="sc-EHOje dhIMBA"
  >
    <pre
      style={
        Object {
          "background": "#F0F0F0",
          "color": "#444",
          "display": "block",
          "margin": 0,
          "overflowX": "auto",
          "padding": "0.5em",
        }
      }
    >
      <code>
        curl \\
    -H 
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          "Authorization: Bearer 
          <span
            style={
              Object {
                "color": "#BC6060",
              }
            }
          >
            \${JWT}
          </span>
          "
        </span>
         \\
    -H 
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          "Content-type: application/json"
        </span>
         \\
    --request POST \\
    --data 
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          '{
      "template_id": "f33517ff-2a88-4f6e-b855-xxxxxxxxxxxx",
      "email_address": "lisa@example.com",
      "personalisation": {
        "user_name": "Lisa",
        "amount_owing": "$15.50"
      }
    }'
        </span>
         \\
    https://rest-api.notify.gov.au/v2/notifications/email

      </code>
    </pre>
  </div>
  <div
    className="sc-EHOje vChbq"
  >
    <pre
      style={
        Object {
          "background": "#F0F0F0",
          "color": "#444",
          "display": "block",
          "margin": 0,
          "overflowX": "auto",
          "padding": "0.5em",
        }
      }
    >
      <code>
        HashMap&lt;String, String&gt; personalisation = 
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          new
        </span>
         HashMap&lt;String, String&gt;();
personalisation.put(
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          "user_name"
        </span>
        , 
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          "Lisa"
        </span>
        );

SendEmailResponse response = client.sendEmail(
    
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          "f33517ff-2a88-4f6e-b855-xxxxxxxxxxxx"
        </span>
        ,
    
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          "lisa@example.com"
        </span>
        ,
    personalisation,
    
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          "your-unique-reference-id"
        </span>
        );

      </code>
    </pre>
  </div>
  <div
    className="sc-EHOje vChbq"
  >
    <pre
      style={
        Object {
          "background": "#F0F0F0",
          "color": "#444",
          "display": "block",
          "margin": 0,
          "overflowX": "auto",
          "padding": "0.5em",
        }
      }
    >
      <code>
        notifications_client.send_email_notification(
    email_address=
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          'lisa@example.com'
        </span>
        ,
    template_id=
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          'f33517ff-2a88-4f6e-b855-xxxxxxxxxxxx'
        </span>
        ,
    personalisation={
        
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          'user_name'
        </span>
        : 
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          'Lisa'
        </span>
        ,
        
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          'amount_owing'
        </span>
        : 
        <span
          style={
            Object {
              "color": "#880000",
            }
          }
        >
          '$15.50'
        </span>
        
    }
)

      </code>
    </pre>
  </div>
</div>
`;
